css_tasks/task10/js/form.js

```diff
@@ -0,0 +1,157 @@
+import {constants} from './constants.js';
+import {DOMElements} from './elements.js';
+
+const {
+  nameText,
+  namePattern,
+  emailText,
+  emailPattern,
+  passwordPattern,
+  passwordText,
+  successText,
+} = constants;
+const {
+  formInputContainer,
+  inputs,
+  signupForm,
+  formButton,
+  formText,
+} = DOMElements;
+
+const submitForm = (e) => {
+  e.preventDefault();
+  validateName(e.target);
+  validateEmail(e.target);
+  validatePassword(e.target);
+  showSuccessMessage();
+};
+
+const validateName = (formName) => {
+  const name = formName.elements.name;
+  const nameValue = name.value;
+  hideWarning(name);
+  name.addEventListener('focus', () => removeDisabledState(formButton));
+  if (!namePattern.test(nameValue)) {
+    handleError(name, nameText, formButton);
+  }
+};
+
+//  Validate Email Input
+
+const validateEmail = (formName) => {
+  const email = formName.elements.email;
+  const emailValue = email.value;
+  hideWarning(email);
+  email.addEventListener('focus', () => removeDisabledState(formButton));
+  if (!emailPattern.test(emailValue)) {
+    handleError(email, emailText, formButton);
+  }
+};
+
+// Validate Password Input
+
+const validatePassword = (formName) => {
+  const password = formName.elements.password;
+  const passwordValue = password.value;
+  password.addEventListener('focus', () => removeDisabledState(formButton));
+  hideWarning(password);
+  if (!passwordPattern.test(passwordValue)) {
+    handleError(password, passwordText, formButton);
+  }
+};
+
+// Show warning
+
+const showWarning = (formName, text) => {
+  const warningMessage = document
+      .createRange()
+      .createContextualFragment(`<p class="form-error">${text}</p>`);
+  formName.after(warningMessage);
+};
+
+// Hide warning message if the input passes validation
+
+const hideWarning = (elem) => {
+  const warning = elem.nextElementSibling;
+  if (warning && warning.classList.contains('form-error')) {
+    warning.remove();
+  }
+};
+
+// Add disabled attribute to the button
+
+const addDisabledState = (elem) => {
```

Comment:
probably you shouldn't rely on class on *another* element, it's unintuitive and required traversing each time. You can store some info about element using data attributes